packages/mobile/src/components/skeleton/Skeleton.tsx

```diff
@@ -20,48 +19,46 @@ type SkeletonProps = {
   noShimmer?: boolean
 }
 
-const createStyles = (themeColors: ThemeColors) =>
-  StyleSheet.create({
-    view: {
-      position: 'relative',
-      overflow: 'hidden',
-      borderRadius: 4
-    },
-    skeleton: {
-      position: 'absolute',
-      width: '400%',
-      height: '100%',
-      backgroundColor: themeColors.skeleton
-    }
-  })
+const useStyles = makeStyles(({ palette }) => ({
+  view: {
+    position: 'relative',
+    overflow: 'hidden',
+    borderRadius: 4
+  },
+  skeleton: {
+    position: 'absolute',
+    width: '400%',
+    height: '100%',
+    backgroundColor: palette.skeleton
+  }
+}))
 
 const Skeleton = ({ width, height, style, noShimmer }: SkeletonProps) => {
-  const styles = useThemedStyles(createStyles)
-  const [shimmerWidth, setShimmerWidth] = useState(0)
-  const [shimmerPos, setShimmerPos] = useState(new Animated.Value(0))
+  const styles = useStyles()
+  const shimmerPos = useRef(new Animated.Value(0)).current
   const { skeleton, skeletonHighlight } = useThemeColors()
 
-  useEffect(() => {
-    if (shimmerWidth !== 0 && !noShimmer) {
+  const shimmer = useMemo(
+    () =>
       Animated.loop(
         Animated.timing(shimmerPos, {
           toValue: 0,
           duration: ANIMATION_DURATION_MS,
           useNativeDriver: true,
           easing: Easing.ease
         })
-      ).start()
-    }
-  }, [shimmerPos, shimmerWidth, noShimmer])
+      ),
+    [shimmerPos]
+  )
 
   return (
     <View style={[styles.view, { height, width }, style]}>
       <Animated.View
         onLayout={(e) => {
           if (noShimmer) return
           const { width } = e.nativeEvent.layout
-          setShimmerWidth(width)
-          setShimmerPos(new Animated.Value(-0.75 * width))
+          shimmerPos.setValue(-0.75 * width)
+          shimmer.start()
```

Comment:
smart so does this prevent a rerender or just speed up when we start to shimmer?

Reply:
Prevents a render bc we aren't setting width in state